fix: use new TranslationDialog component

edoardo opened the pull request:
**Requires https://github.com/dhis2/analytics/pull/1342**

Replace the old `d2-ui` component with the new one from analytics.

### Screenshot
<img width="1167" alt="Screenshot 2022-09-22 at 13 58 15" src="https://user-images.githubusercontent.com/150978/191740746-1138edf5-c99d-4597-83a7-a1f79aded383.png">

edoardo committed:
fix: use new TranslationDialog component

edoardo committed:
chore: fix lint error

jenniferarnesen commented on src/pages/edit/ActionsBar.js:
```diff
@@ -159,23 +157,13 @@ const EditBar = ({ dashboard, ...props }) => {
-                }
                 onTranslationSaved={Function.prototype}
-                insertTheme={true}
-                isOnline={online}
```
Does the new dialog support the offline disabling of the Save/Cancel buttons?

edoardo replied:
Good point.
I don't think the current implementation of `TranslationDialog` has offline support, but it won't be difficult to add.
I think the `useOnlineStatus` hook can be used directly in the component in `analytics`.

edoardo replied:
Added offline support for `TranslationDialog` in `analytics`.
Because the `OfflineTooltip` component is needed there, it's been moved from the app to `analytics` and used from there.

jenniferarnesen commented:
Could you add a screenshot to the description?

jenniferarnesen commented:
Remember to yarn remove package d2-ui-translation-dialog

edoardo committed:
refactor: remove component moved to analytics

edoardo committed:
refactor: use shared OfflineTooltip component from analytics

edoardo committed:
chore: remove TranslationDialog component from old d2-ui

This has been replaced with the new component in analytics

edoardo committed:
fix: use OfflineTooltip from analytics

edoardo committed:
test: fix tests and snapshots

edoardo committed:
chore: bump analytics for OfflineTooltip and TranslationDialog

jenniferarnesen commented:
I found a bug. To reproduce:
1. Go into dashboard edit mode and click "Translate" (new translation dialog opens).
2. In dev tools Network tab, turn off network ("No throttling" -> "Offline").
3. An error appears in the alert bar at the bottom

![image](https://user-images.githubusercontent.com/6113918/194360646-75f145a4-69c6-4544-b205-2cf7d4fd932c.png)

![image](https://user-images.githubusercontent.com/6113918/194360776-543976a7-e166-44fe-a24f-bf22e021edbf.png)

jenniferarnesen reviewed (changes requested):
See screenshots of error

edoardo committed:
chore: update @dhis2/analytics to solve compile issue

edoardo committed:
fix: avoid rerender which triggers unnecessary request

This fixes an issue where when going offline/online a request to fetch
the translation was triggered resulting in an error when offline.
The request is unnecessary anyway, since if online it happens anyway,
and when offline does not make sense since it can only fail without
network.

edoardo committed:
test: update snapshots
